EventHubs/tests/test_eventhub_metrics.py

```diff
@@ -1,48 +1,101 @@
-        self.insert_mock_logs_in_EventHub('metrics_fixtures.json')
-        self.check_error_logs()
+        self.assertTrue(self.resource_group_exists(self.RESOURCE_GROUP_NAME)) 
+        self.insert_mock_metrics_in_EventHub('metrics_fixtures.json')
+        time.sleep(300)  # Due to latency, logs are available after few mins.
+        self.check_success_log()
+        self.check_error_log()
+        self.check_warning_log()
+    
+    def fetchlogs(self, query):
+        original_stdout = sys.stdout
+        sys.stdout = StringIO()
 
-    def check_error_logs(self):
-        print("sleeping 1min for function execution")
-        query = "PartitionKey eq 'R2'"
-        self.wait_for_table_results(query)
-        self.assertTrue(self.get_row_count(query) > 0)
+        try:
+            client = LogsQueryClient(self.azure_credential)
+            response = client.query_workspace(self.get_Workspace_Id(), query, timespan=timedelta(minutes=15))
+                
+            if response.status == LogsQueryStatus.PARTIAL:
+                error = response.partial_error
+                data = response.partial_data
+            elif response.status == LogsQueryStatus.SUCCESS:
+                data = response.tables
+            for table in data:
+                for col in table.columns:
+                    print(col + "    ", end="")
+                for row in table.rows:
+                    for item in row:
+                        print(item, end="")
+                    print("\n")
+        except Exception as e:
+            print("An unexpected error occurred during the test:")
+            print("Exception", e)
 
-        rows = self.table_service.query_entities(
-            self.log_table_name, filter=query)
+        output = sys.stdout.getvalue()
 
-        haserr = False
-        for row in rows.items:
-            print("LogRow: ", row["FunctionName"], row["HasError"])
-            if row["FunctionName"].startswith(self.function_name_prefix) and row["HasError"]:
-                haserr = True
+        sys.stdout = original_stdout
+
+        return output
+    
+    def check_success_log(self):
+        app_insights = self.get_resource('Microsoft.Insights/components')
+        query = f"union *, app('{app_insights.name}').traces | where message == 'Sent all metric data to Sumo. Exit now.' | where operation_Name == 'EventHubs_Metrics' | where customDimensions.Category == 'Function.EventHubs_Metrics.User'"
+        captured_output = self.fetchlogs(query)
+        haslog = False
+        if ('Sent all metric data to Sumo. Exit now.' in captured_output):
+            haslog = True
+        self.assertTrue(haslog)
 
+    def check_error_log(self):
+        app_insights = self.get_resource('Microsoft.Insights/components')
+        query = f"union *, app('{app_insights.name}').traces | where message == '""LogLevel"":""Error""' | where operation_Name == 'EventHubs_Metrics' | where customDimensions.Category == 'Function.EventHubs_Metrics.User'"
```

Comment:
1. we can use severityLevel = 2 and severityLevel = 3 for filtering warning and error messages. ust to avoid hardcoding. 
app('{app_insights.name}').traces  | where operation_Name == 'EventHubs_Metrics and severityLevel == 2 
2. In assertTrue print the captured output as well otherwise we won't know what is the error which occurred
3. Why do we require  customDimensions.Category in where clause?

Reply:
updated